core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -624,6 +683,38 @@ class CityInfo {
         civInfo.updateDetailedCivResources() // this building could be a resource-requiring one
     }
 
+    private fun demandNewResource() {
+        val candidates = getRuleset().tileResources.values.filter {
+            it.resourceType == ResourceType.Luxury && // Must be luxury
+            !it.hasUnique(UniqueType.CityStateOnlyResource) && // Not a city-state only resource eg jewelry
+            it.name != demandedResource && // Not same as last time
+            !civInfo.hasResource(it.name) && // Not one we already have
+            it.name in tileMap.resources && // Must exist somewhere on the map
+            getCenterTile().getTilesInDistance(3).none { nearTile -> nearTile.resource == it.name } // Not in this city's radius
+        }
+
+        val chosenResource = candidates.randomOrNull()
+        /* What if we had a WLTKD before but now the player has every resource in the game? We can't
+           pick a new resource, so the resource will stay stay the same and the city will demand it
+           again even if the player still has it. But we shouldn't punish success. */
+        if (chosenResource != null)
+            demandedResource = chosenResource.name
+        if (demandedResource == "") // Failed to get a valid resource, try again some time later
+            setFlag(CityFlags.ResourceDemand, 15 + Random().nextInt(10))
```

Comment:
So if you found a new city after having all luxuries, it will never celebrate we love the king day?

Reply:
Yes, and I wasn't able to test this in-game in civ V, but from CvCity.cpp it is the behavior.
It is slightly inelegant, I'll admit. Other thinkable solutions:

1. A new city forces a luxury to be picked, making the behavior similar to old cities.
2. Once all luxuries are controlled, no cities demand more luxuries and so there is no WLTKD. This is perhaps the most intuitive solution, but it is punishing success, and encourages gamey behavior like intentional pillaging or breaking trades to get new WLTKD:s.
3. As long as all luxuries are controlled, all cities enter a permanent WLTKD of sorts without demanding any particular resource. Mainly a cosmetic difference from option 1, might reduce notification spam in the late game.

Reply:
I don't think it matters all that much, as having all resources under your control almost always means you have basically won the game anyways (though excessive trading might be exception to this). In my personal opinion, option 3 would make the most sense as long as this behaviour is explained in the tutorial section on WLTKD. This also incentives trading for resources more than just the +4 happiness, as that often doesn't matter all that much.